wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +37,380 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
     LockedWallet(nodeApi, chainQueryApi)
   }
 
+  /** We need a output so we can get proper tx size estimation */
+  private def getDummyDLCFundingOutputs(
+      amount: CurrencyUnit): Vector[TransactionOutput] = {
+    val key = ECPublicKey.freshPublicKey
+    val spk = P2WSHWitnessSPKV0(MultiSignatureScriptPubKey(2, Seq(key, key)))
+
+    // funded output and the other party's change
+    Vector(TransactionOutput(amount, spk), TransactionOutput(Satoshis.one, spk))
+  }
+
+  override def createDLCOffer(
+      amount: Bitcoins,
+      oracleInfo: OracleInfo,
+      contractInfoVec: Vector[Sha256DigestBE],
+      feeRateOpt: Option[SatoshisPerVirtualByte],
+      locktime: UInt32,
+      refundLocktime: UInt32): Future[DLCOffer] = {
+    logger.debug("Calculating relevant wallet data for DLC Offer")
+
+    val contractInfo =
+      ContractInfo(
+        (Map.newBuilder ++= contractInfoVec.map(sha => (sha, amount.satoshis)))
+          .result())
+
+    val timeouts = DLCTimeouts(5,
+                               BlockStamp(locktime.toInt),
+                               BlockStamp(refundLocktime.toInt))
+
+    val feeRate = feeRateOpt.getOrElse(SatoshisPerVirtualByte.one)
+
+    val eventId = DLCMessage.calcEventId(oracleInfo, contractInfo, timeouts)
+
+    logger.debug("Checking if offer has already been made")
+    for {
+      account <- getDefaultAccountForType(AddressType.SegWit)
+      dlcDbOpt <- dlcDAO.findByEventId(eventId)
+      offer <- dlcDbOpt match {
+        case Some(dlcDb) =>
+          logger.debug("DLC Offer has already been made, returning offer")
+          listUtxos(dlcDb.fundingInputs).map { spendingInfoDbs =>
+            val fundingInputs =
+              spendingInfoDbs
+                .map { db =>
+                  val info =
+                    db.toUTXOSpendingInfo(account, keyManager, dlcDb.network)
+                  (info.outPoint, info.output)
+                }
+            DLCOffer(
+              dlcDb.contractInfo,
+              oracleInfo,
+              keyManager.getRootXPub, // fixme this will need to be changed according to KeyDerivation.md
+              amount.satoshis,
+              fundingInputs,
+              Bech32Address(dlcDb.changeSPK, dlcDb.network),
```

Comment:
probably not the right place to ask this question, but `dlcDb.changeSpk` is tracked as part of a normal wallet key chain right?